### PR TITLE
Add C-style null compare to decompiler

### PR DESCRIPTION
**Pull request by intranautic**
Changes all decompiler expressions like so:
- `x == 0` => `!x`
- `x != 0` => `x`

### CODE REVIEW

File: angr/analyses/decompiler/structured_codegen/c.py [hunk 1]
```diff
@@ -1567,22 +1567,24 @@ class CBinaryOp(CExpression):
         "rhs",
         "tags",
         "common_type",
+        "_cstyle_null_cmp"
     )
 
     def __init__(self, op, lhs, rhs, tags: Optional[dict] = None, **kwargs):
         super().__init__(**kwargs)
-
```
Comment (mahaloz) [inline angr/analyses/decompiler/structured_codegen/c.py:1574]: Add this space back

File: angr/analyses/decompiler/structured_codegen/c.py [hunk 2]
```diff
@@ -1567,22 +1567,24 @@ class CBinaryOp(CExpression):
...
         self.tags = tags
+        self._cstyle_null_cmp = self.codegen.cstyle_null_cmp
 
         self.common_type = self.compute_common_type(self.op, self.lhs.type, self.rhs.type)
         if self.op.startswith("Cmp"):
             self._type = SimTypeChar().with_arch(self.codegen.project.arch)
         else:
             self._type = self.common_type
 
+
```
Comment (mahaloz) [inline angr/analyses/decompiler/structured_codegen/c.py]: Remove this space

File: angr/analyses/decompiler/structured_codegen/c.py [hunk 3]
```diff
@@ -1711,18 +1726,21 @@ def _c_repr_chunks(self, op):
...
-            1 if self.op in ["Sub", "Div"] else 0
-        ):
-            paren = CClosingObject("(")
-            yield "(", paren
-            yield from self._try_c_repr_chunks(self.rhs)
-            yield ")", paren
-        else:
-            yield from self._try_c_repr_chunks(self.rhs)
+
+        # if there is a null compare then neglect to yield operator and right hand side
```
Comment (mahaloz) [inline angr/analyses/decompiler/structured_codegen/c.py:1720]: comment not needed here

**Review by mahaloz — CHANGES_REQUESTED** [review]

**Comment by mahaloz** [comment]
LGTM, just waiting on CI

**Comment by mahaloz** [comment]
@ltfish some CI fails now due to this text change:
https://github.com/angr/angr/actions/runs/5206392860/jobs/9393011392#step:5:166

There are two options:
1. We can turn off `c-style null compare` by default
2. Change all testcases that depend on the string ` == 0` and `  != 0`

I would prefer the second option, but lmk what you think.

**Comment by ltfish** [comment]
@mahaloz please change/fix all impacted test cases.

**Pull request merged**